#!/usr/bin/env python
PACKAGE = "fub_mpc_planner"

from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()

gen.add("delta_t",    double_t,    0, "delta time s", 0.1,  0, 10)
gen.add("vx",    double_t,    0, "longitudinal velocity m/s", 100,  0, 200)
gen.add("vy",    double_t,    0, "lateral velocity m/s", 0,  0, 200)
gen.add("vx_min",    double_t,    0, "minimum longitudinal velocity m/s", 0,  0, 30)
gen.add("vx_max",    double_t,    0, "maximum longitudinal velocity m/s", 100,  0, 200)
gen.add("vy_min",    double_t,    0, "minimum lateral velocity m/s", -20,  -100, 100)
gen.add("vy_max",    double_t,    0, "maximum lateral velocity m/s", 20,  -100, 100)
gen.add("ax_min",    double_t,    0, "minimum longitudinal accelaration m/s2", -80,  -200, 0)
gen.add("ax_max",    double_t,    0, "maximum longitudinal accelaration m/s2", 80,  0, 200)
gen.add("ay_min",    double_t,    0, "minimum lateral accelaration m/s2", -20,  -100, 0)
gen.add("ay_max",    double_t,    0, "maximum lateral accelaration m/s2", 20,  0, 100)
gen.add("left_boundry_y",    double_t,    0, "left lateral boundry m", -6,  -100, 100)
gen.add("right_boundry_y",    double_t,    0, "right lateral boundry m",2,  -100, 100)
gen.add("vehicle_width",    double_t,    0, "vehicle width m", 2,  0, 100)

# street_type_enum = gen.enum([ gen.const("ROUNDABOUT",      int_t, 0, "round street"),
#                        gen.const("HIGHWAY",     int_t, 1, "HIGHWAY")],
#                      "An enum to set street type")

# gen.add("type", int_t, 0, "street type", 1, 0, 3, edit_method=street_type_enum)


exit(gen.generate(PACKAGE, "fub_mpc_planner", "param"))
